### PR TITLE
[beta-updates] linting_tool

### PR DESCRIPTION
**Pull request by craiglabenz**
Updated linting_tool for Dart 2.17

### CODE REVIEW

File: experimental/linting_tool/lib/app.dart [hunk 1]
```diff
@@ -19,10 +19,10 @@ class LintingTool extends StatefulWidget {
   static const String homeRoute = routes.homeRoute;
 
   @override
-  _LintingToolState createState() => _LintingToolState();
+  LintingToolState createState() => LintingToolState();
```
Comment (domesticmouse) [inline experimental/linting_tool/lib/app.dart:22]: `State<LintingTool> createState() => _LintingToolState();`

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: experimental/linting_tool/lib/app.dart (hunk 1 above)
Comment (craiglabenz) [reply experimental/linting_tool/lib/app.dart]: Updated all instances in this PR

File: experimental/linting_tool/lib/widgets/lint_expansion_tile.dart [hunk 2]
```diff
@@ -268,14 +274,19 @@ class _NewProfileDialog extends StatelessWidget {
...
     required this.rule,
   }) : super(key: key);
 
   final Rule rule;
 
+  @override
+  State<ExistingProfileDialog> createState() => ExistingProfileDialogState();
+}
+
+class ExistingProfileDialogState extends State<ExistingProfileDialog> {
```
Comment (domesticmouse) [inline experimental/linting_tool/lib/widgets/lint_expansion_tile.dart:289]: This one probably needs hiding. I think.

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: experimental/linting_tool/lib/widgets/lint_expansion_tile.dart (hunk 2 above)
Comment (craiglabenz) [reply experimental/linting_tool/lib/widgets/lint_expansion_tile.dart]: whoops. fixt.

**Pull request merged**